src/index.js

```diff
@@ -1,25 +1,34 @@
 const isNullOrUndefined = require('util').isNullOrUndefined;
 const express = require('express');
 const bodyParser = require('body-parser');
-const app = express();
 
+const ConversationDelegate = require('./conversation/ConversationDelegate');
+const ConversationRouter = require('./conversation/ConversationRouter');
+
+const app = express();
 app.use(bodyParser.json());
 
+/* configure all of the things */
 const BotApi = require('./api/BotApi');
-const ConversationRouter = require('./conversation/ConversationRouter');
 
-/* configure all of the things */
-const botApi = new BotApi();
+const conversationRouter = new ConversationRouter(app);
+const saveReadingDelegate = new ConversationDelegate(app, 'saveReading');
+conversationRouter.registerConversationDelegate(saveReadingDelegate);
 
+const botApi = new BotApi(app);
+app.set('config', {
+  botApi: botApi,
+  conversationRouter: conversationRouter,
+});
 
 app.get('/message', function (req, res) {
```

Comment:
No, this is a way to set some shared configuration across the application - just storing a dict. But yeah, it does like pretty similar to an actual request